nixos/modules/services/databases/mysql.nix

```diff
@@ -180,12 +194,14 @@ in
         preStart =
           ''
             if ! test -e ${cfg.dataDir}/mysql; then
-                mkdir -m 0700 -p ${cfg.dataDir}
-                chown -R ${cfg.user} ${cfg.dataDir}
+                mkdir -p ${cfg.dataDir}
                 ${mysql}/bin/mysql_install_db ${mysqldOptions}
                 touch /tmp/mysql_init
             fi
 
+            chmod -m 700 ${cfg.dataDir}
+            chown -R ${cfg.user} ${cfg.dataDir}
```

Comment:
It's possible to ask chmod/chown to only update permissions if they differ, is that better?